samples/OriginalUriBaseIds.sarif

```diff
@@ -0,0 +1,77 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "properties": {

```

Comment:
>properties [](start = 3, length = 10)

btw - original url base ids shouldn't create 'loops'. since we have a dedicated 'don't screw up original uri base ids' rule, we could add this verification to that check.

Reply:
We're about to have one of our partitioning discussions. Rule SARIF1018 is about the validity of the URIs in `originalUriBaseIds`: They have to end with a slash, and if there's no `uriBaseId`, they have to be absolute. The (good) rule you're proposing is about the structure of the `uriBaseIds`. I'll make it part of 1018 if you want. But do consider how you would (1) friendly-name, and (2) describe, the combined rule.

---
In reply to: [414814948](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414814948) [](ancestors = 414814948)

Reply:
I filed [https://github.com/microsoft/sarif-sdk/issues/1864](https://github.com/microsoft/sarif-sdk/issues/1864), "Analysis rule: uriBaseIds should not form loops". In the description I explicitly say that we haven't decided whether this is a separate rule or an enhancement to an existing one.

---
In reply to: [414865952](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414865952) [](ancestors = 414865952,414814948)